File: problems/1.two-sum.rb
#
# @lc app=leetcode id=1 lang=ruby
#
# [1] Two Sum
#

# @lc code=start
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  # brute-force exceeds time limits, use Hash.
  h = {}
  for i in 0..nums.length-1 do
    h[nums[i].to_s] = i
  end

  for i in 0..nums.length-1 do
    rest = target - nums[i]
    j = h[rest.to_s]
    if j != nil && i != j then
      return [i, j]
    end
  end
end
# @lc code=end

File: problems/1.two-sum.c
/*
 * @lc app=leetcode id=1 lang=c
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
  int i, j;
  int* ret;
  ret = (int*)malloc(sizeof(int) * 2);
  *returnSize = 2;
  for(i=0; i<numsSize; i++){
    for(j=i+1; j<numsSize; j++){
      if(nums[i] + nums[j] == target){
        ret[0] = i;
        ret[1] = j;
        return ret;
      }
    }
  }
  return ret;
}
// @lc code=end

